### PR TITLE
Dev magda starting to apply UI in questions tab

### PR DESCRIPTION
**Pull request by eynimeni**
#50

### CODE REVIEW

File: classes/catquiz.php [hunk 1]
```diff
@@ -136,13 +136,7 @@ public static function return_sql_for_catscalequestions(
...
         global $DB;
         if ($contextid === 0) {
-            $default_context = $DB->get_record_sql(
-                'SELECT id FROM {local_catquiz_catcontext} WHERE '.$DB->sql_like('json', ':default'),
-                [
-                    'default' => '%"default":true%',
-                ]
-            );
-            $contextid = $default_context->id;
+            $contextid = self::get_default_context_id();;
```
Comment (davidszkiba) [inline classes/catquiz.php:139]: There's an extra `;`

File: classes/form/scaleselector.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        global $CFG, $DB, $PAGE;
+
+        $mform = $this->_form;
+        $data = (object) $this->_ajaxformdata;
+        $customdata = $this->_customdata;
+
+        $type = isset($customdata['type'])? $customdata['type'] : "scale"; //Type i.e. 'subscale'.
+
+
+        // $default = get_string('searchcatcontext', 'local_catquiz'); TODO default value
```
Comment (davidszkiba) [inline classes/form/scaleselector.php:56]: I guess that can be removed, or?

File: classes/form/scaleselector.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        $sql = "SELECT $select FROM $from WHERE $where";
+
+        $contextsdb = $DB->get_records_sql($sql, $params);
+        $contexts = [];
+        foreach ($contextsdb as $contextid => $context) {
+            $contexts[$contextid] = $context->name;
+        }
+        $options = array(
+            'multiple' => false,
+            //'noselectionstring' => $default,
```
Comment (davidszkiba) [inline classes/form/scaleselector.php:80]: I think this can be removed?

File: classes/form/scaleselector.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        );
+
+        $label = '';
+
+        if(isset($customdata['label'])) {
+            $label = $customdata['label'];
+        }
+
+        $mform->addElement('select', $type, $label, $contexts, $options);
+        //$mform->setDefault('subscale', $default);
```
Comment (davidszkiba) [inline classes/form/scaleselector.php:91]: Can be removed?

File: classes/output/questionsdisplay.php [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        $catcontext = empty($this->catcontextid) ? catquiz::get_default_context_id() : $this->catcontextid; // If no context is set, get default context from DB.
+
+        $table = new catscalequestions_table('questionstable', $this->tablescale, $catcontext);
+
+        list($select, $from, $where, $filter, $params) = catquiz::return_sql_for_catscalequestions([$this->tablescale], $catcontext, [], []);
+
+        $table->set_filter_sql($select, $from, $where, $filter, $params);
+
+        $columnsarray = [
+            //'id' => get_string('id', 'core'), // TODO what is label?
```
Comment (davidszkiba) [inline classes/output/questionsdisplay.php:183]: Can this be removed?

File: classes/output/questionsdisplay.php [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+            ]
+        ];
+        $table->placebuttonandpageelementsontop = true;
+
+        $table->tabletemplate = 'local_wunderbyte_table/twtable_list';
+        $table->define_cache('local_catquiz', 'testitemstable');
+
+        $table->pageable(true);
+
+        //$table->infinitescroll = 7;
```
Comment (davidszkiba) [inline classes/output/questionsdisplay.php:246]: Can be removed?

File: classes/table/catscalequestions_table.php [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+
+    /**
+     * Overrides the output for this column.
+     * @param object $values
+     */
+    public function col_name($values) {
+
+        global $OUTPUT;
+
+        //try {
```
Comment (davidszkiba) [inline classes/table/catscalequestions_table.php:213]: Can this be removed?

File: classes/table/catscalequestions_table.php [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+        switch ($values->maxstatus) {
+            case 5:
+                $color = 'green';
+                break;
+            case 1:
+                $color = 'orange';
+                break;
+            case 0:
+                $color = 'red';
```
Comment (davidszkiba) [inline classes/table/catscalequestions_table.php:153]: Lets take yellow here

File: classes/table/catscalequestions_table.php [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+                break;
+            case -5:
```
Comment (davidszkiba) [inline classes/table/catscalequestions_table.php:155]: .. and red here

Comment (eynimeni) [reply classes/table/catscalequestions_table.php:155]: is changed

**Review by davidszkiba — APPROVED** [review]
:+1:

**Pull request merged**